File: Scala/src/main/scala/Recursion.scala
package main.scala

object Recursion {
  def factorial(n: Int): Int =  {
    if(n == 1){
      n
    } else {
      n * factorial(n - 1)
    }
  }
  def main(args: Array[String]){
    val sc = scala.io.StdIn.readInt()
    val result = factorial(sc);
    println(result)
  }
}

File: Java/src/com/hackerrank/daysofcode/Recursion.java
package com.hackerrank.daysofcode;
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;
/**
 * Created by vmurakami on 25/07/17.
 */
public class Recursion {

    static int factorial(int n) {
        if(n == 1){
            return n;
        } else {
            return n * factorial(n - 1) ;
        }
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int result = factorial(n);
        System.out.println(result);
    }
}
